app/src/Vues/AddRole.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState, useEffect } from "react";
+import AddRoleCard from "../Components/AddRole/AddRoleCard";
+import Notification from "../Components/AddRole/AddRolePopUp";
+import UserApi from "../scripts/API/ModelAPIs/UserAPI";
+import { UserType } from "../scripts/API/APITypes/Users.ts";
+import APIResponse from "../scripts/API/Responses/APIResponse.ts";
+
+const rolesList = [
```

Comment:
Maybe consider moving this array into an API class.
You can ask me if you want more informations.